test/subset-band-name/utilities.py

```diff
@@ -0,0 +1,118 @@
+""""
+Common utility functions used by the subset-band-name regression tests.
+"""
+
+import os
+from pyhdf.SD import SD, SDC
+import numpy
+
+from harmony import Client, Request
+from harmony.harmony import ProcessingFailedException
+
+
+def submit_and_download(
+    harmony_client: Client, request: Request, file_indicator: str
+) -> str:
+    """Submit a Harmony request via a `harmony-py` client. Wait for the
+    Harmony job to finish, then download the results to the specified file
+    path.
+
+    """
+    downloaded_filenames = []
+    output_filename = None
+
+    try:
+        job_id = harmony_client.submit(request)
+
+        print(f"Job ID: {job_id}")
+
+        for filename in [
+            file_future.result()
+            for file_future in harmony_client.download_all(job_id, overwrite=True)
+        ]:
+
+            print(f'Downloaded: {filename}')
+            downloaded_filenames.extend([filename])
+
+            # make a copy to /workdir/output (debugging)
+            # output_filename = f'/workdir/output/{filename.split("/")[-1]}'
+            # os.system(f'cp {filename} {output_filename}')
+
+        for filename in downloaded_filenames:
+            if file_indicator in filename:
+                output_filename = filename
+                print(f'Saved output to: {output_filename}')
+            else:
+                print(f'Could not save output to: {filename}')
+
+    except ProcessingFailedException as exception:
+        print_error('Harmony request failed to complete successfully.')
+        raise exception
+
+    return output_filename
+
+
+def get_dim_sizes(file: str) -> list[int]:
```

Comment:
I don't think this function is called anywhere. It could probably be removed.

Reply:
removed